src/pages/Governance/Proposal/card/VoteButtons.tsx

```diff
@@ -110,7 +111,7 @@ export default function VoteButtons({proposalId}: Props) {
         action={voteOnSuccessSnackbarAction}
       >
         {`Vote succeeded with transaction ${
-          (voteResponse as VoteResponseOnSuccess).transactionHash
+          (transactionResponse as TransactionResponseOnSuccess).transactionHash
```

Comment:
It should be possible to omit the `as TransactionResponseOnSuccess` cast (possibly in combination with the change to `succeeded: true / false` mentioned above). https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing It's best to avoid using `as` because it can cause things to break at runtime (https://dev.to/ylerjen/typescript-cast-is-a-type-breaker-1jbh has more details).

Reply:
I tried to change to `succeeded: true / false` - that doesnt seem to help. We need to specify what type to use since this is a union type and not all types have the same props. (looks like the article you attached is also ok with precising the type from union types)